Makefile

```diff
@@ -78,3 +78,7 @@ endef
 check-jq:
 	@printf "Checking if jq is installed... "
 	@command -v jq >/dev/null 2>&1 && { printf "OK\n"; } || { printf "FAILED. Please install jq to proceed.\n"; exit 1; }
+
+.PHONY: sonic
+sonic:
+	go build -tags=sonic ./...
```

Comment:
This is different from all previous examples.
You're not building sonic. You're building gct with sonic.
And it leaves a gap where install should be.

I suggest that users need to use `make build` and `install` with an argument:
```
make install JSON=sonic
```

Or, if we go the route I'm proposing:
```
make install JSON=golang
```

Also: Given the recent highlighting of `act`, I wonder if we remove Makefile and update docs to use `act`, and make that a (non-runtime) dependency for installing GCT from source.
Note: As we approach semver (right?), we should be issuing binaries anyway

Reply:
If we are going to allow customisation, I think a more general approach `make build FLAGS="-tags=sonic"` might be better. This gives users the flexibility to specify any build tags or flags they need without hardcoding specific options in the Makefile. If users have to manually specify flags every time, they might as well use go build or go install directly. The goal of make is to simplify workflows, not add unnecessary complexity.

We could:

Keep the Makefile but allow FLAGS to be overridden for advanced use cases.
Provide default targets like make build-sonic for common configurations.

As for act that is an option. But I have little to no experience in semver and issuing binaries so you can discuss with @thrasher- && @gloriousCode

Reply:
If you're going to use FLAGS, then:
```
GOFLAGS="-tags=hedgehogless" make build
```
makes more sense than adding it to the end, and means no extra support for FLAGS in Makefile.
Note: I'll accept nosonic or sonicless )

act: I've only just started using it

Semver is a separate topic, and issuing binaries doesn't need to be related to it, now that I think about it.

Reply:
1) Can keep the Makefile, as I find other recipes useful locally like pprof and only use act for simulating GHA workflows which I don't think it can do that also. We also want to keep the number of dependencies (runtime and non-runtime) down as much as possible for GCT ouf of the box. 
2) I'm happy with supporting both options who want a quick shorthand recipe and make install/make build param for power users (we do this with LTC where things are diisabled by NO_X).